### ISSUE TITLE
Autofix remplace the char it supposed to suffix.

### ISSUE THREAD

**Issue by michaeljota** [Microsoft/vscode-tslint#121]
Hi! I'm having a bug when trying to run autofix. The last character in the line, it's being remplace with the fix. Giving:

```ts
body: {
  id: user.id,
  username: user.username,
  firstName: user.firstName,
  lastName: user.lastName,
  token: 'fake-jwt-token'
}
```
With trailing comma set to true, for example, if I run auto fix in one line, or all the missing commas, or all the file, result in a character being remplace.
```ts
body: {
  id: user.id,
  username: user.username,
  firstName: user.firstName,
  lastName: user.lastName,
  token: 'fake-jwt-token,  // <--- The ' it's gone and the comma got add.
}
```
```ts
body: {
  id: user.id,
  username: user.username,
  firstName: user.firstName,
  lastName: user.lastName,
  token: 'fake-jwt-token'
, // <-- No body closing. 
```

**Comment by rlasjunies** [comment]
Hi @michaeljota ,

 I'v tried to reproduced your issue on my env. without success

 I've added the trailing comma rules 

![image](https://cloud.githubusercontent.com/assets/1439791/20406479/ce632c36-ad0d-11e6-91f0-d2185c143172.png)

Here is the code before apply the autoFix:

![image](https://cloud.githubusercontent.com/assets/1439791/20406517/ef1bb2c2-ad0d-11e6-8ef8-3b4707549451.png)

Here is the result after applying the autoFix:

![image](https://cloud.githubusercontent.com/assets/1439791/20406541/03999caa-ad0e-11e6-9d65-59cae293f43d.png)

Here is my tslint version:
![image](https://cloud.githubusercontent.com/assets/1439791/20406560/167ada5a-ad0e-11e6-84f7-a1c3bec373e6.png)

Can you check your conf?

**Comment by michaeljota** [comment]
tslint.config.json
![config](https://cloud.githubusercontent.com/assets/10507776/20434738/a81263d6-ad7f-11e6-9a08-e6a90143c313.png)

TSLint version:
![version](https://cloud.githubusercontent.com/assets/10507776/20434757/c142805c-ad7f-11e6-8f7f-4c50f5507d6b.png)

Issue:
![autofix](https://cloud.githubusercontent.com/assets/10507776/20434574/b8902a28-ad7e-11e6-87a2-6aa163e88f46.gif)

**Comment by egamma** [comment]
@rlasjunies the comment above shows that you are using tslint 3.5.1. Can you try again with tslint 4.0.1. I suspect that 4.0.1 comes more fixes provided by tslint directly?

**Comment by rlasjunies** [comment]
I suspected the same cause. I tried to work on it this week but I face the same problem than everybody with the release 4.0. 
I'll try it again tonight with your latest version supporting 4.0

Regards

**Comment by Tiscs** [comment]
I just tried with `vscode@1.7.2`, `vscode-tslint@0.6.4` and `tslint@4.0.2`, got the same problem.
``` TypeScript
type stringArray = Array<string>; // before auto fix
```
``` TypeScript
type stringArray = ; // auto fixed
```

**Comment by egamma** [comment]
I have disabled auto fixes that cause problems in the latest version 0.6.5. 

Adding support for these fixes is tracked in #136. Closing this issue.
